pkg/controllers/disruption/consolidation.go

```diff
@@ -198,6 +203,73 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
+//  1. The SpotToSpotConsolidation feature flag is set to true.
+//  2. For single-node consolidation:
+//     a. There are at least 15 cheapest instance type replacement options to consolidate.
+//     b. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate, results *pscheduling.Results,
+	candidatePrice float64) (Command, error) {
+
+	// Spot consolidation is turned off.
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		return Command{}, nil
+	}
+
+	// Since we are sure that the replacement nodeclaim considered for the spot candidates are spot, we will enforce it through the requirements.
+	results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
+	// All possible replacements for the current candidate compatible with spot offerings
+	instanceTypeOptionsWithSpotOfferings :=
+		results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions.Compatible(results.NewNodeClaims[0].Requirements)
+
+	// Possible replacements that are lower priced than the current candidate
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = filterByPrice(instanceTypeOptionsWithSpotOfferings, results.NewNodeClaims[0].Requirements, candidatePrice)
+
+	if len(results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions) == 0 {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "Can't replace spot node with a cheaper spot node")...)
+		}
+		// no instance types remain after filtering by price
+		return Command{}, nil
+	}
+
+	// For multi-node consolidation:
+	// We don't have any requirement to check the remaining instance type flexibility, so exit early in this case.
+	if len(candidates) > 1 {
+		return Command{
+			candidates:   candidates,
+			replacements: results.NewNodeClaims,
+		}, nil
+	}
+
+	// For single-node consolidation:
+	// We check whether we have 15 cheaper instances than the current candidate instance. If this is the case, we know the following things:
+	//   1) The current candidate is not in the set of the 15 cheapest instance types and
+	//   2) There were at least 15 options cheaper than the current candidate.
+	if len(results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions) < MinInstanceTypesForSpotToSpotConsolidation {
+		c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, fmt.Sprintf("SpotToSpotConsolidation requires %d cheaper instance options than the current candidate to consolidate, got %d",
```

Comment:
consider adding type here to be more reflective of the code type too
```suggestion
		c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, fmt.Sprintf("SpotToSpotConsolidation requires %d cheaper instance type options than the current candidate to consolidate, got %d",
```

Reply:
Addressed.